### DISCUSSION TITLE
Restrict use of functions in a kata

### DISCUSSION THREAD

**Discussion by bitcanon** [codewars/codewars.com#2703] [Q&A] [answered]
When writing a kata there are often two options available for solving it:
- Write your own code using the language operators and your own functions
- Use external libraries that does it all for you

I want the user to gain a deeper understanding of a subject when solving the kata. In order to do that the user should use, for example, bitwise operators. The user could also use a built-in function that solves the entire kata in one line, but still have no idea how it works under the hood.

My question is (specifically for the C language), is there a way to prohibit the use of:
- A specific function (for example `pow()`)
- An entire include file (for example `math.h`)

Thanks in advance.

**Comment by Blind4Basics** [comment]
https://docs.codewars.com/authoring/guidelines/submission-tests/#tests-with-additional-restrictions
especially, read the last point, plz

Note: I don't do C myself, so I cannot tell anything more...

**Comment by interrupt-routine** [comment, marked as answer]
You can disable a C function by writing a dummy function with the same name in `setup.c` (preloaded) or `fixture.c`  (tests)
Be careful since most `math.h` functions exist in 3 flavors for `float` ('f' suffix), `double` (no suffix) and long double ('l' suffix) e.g. `powf(), pow(), powl()`
Disabling a header is not very useful, since undeclared functions just emit a warning, and user can just copy-paste the declarations they need anyway.

**Reply by bitcanon** [reply]
Perfect, this is exactly the answer I was looking for. Thanks!
